domain/domain.go

```diff
@@ -44,26 +44,20 @@ type Domain struct {
 	SchemaValidity *schemaValidityInfo
 }
 
-func (do *Domain) loadInfoSchema(txn kv.Transaction) (err error) {
-	defer func() {
-		if err != nil {
-			do.SchemaValidity.setLastFailedTS(txn.StartTS())
-		}
-	}()
-	m := meta.NewMeta(txn)
-	schemaMetaVersion, err := m.GetSchemaVersion()
+func (do *Domain) loadInfoSchema(handle *infoschema.Handle, usedSchemaVersion int64, startTS uint64) error {
+	snapshot, err := do.store.GetSnapshot(kv.NewVersion(startTS))
 	if err != nil {
 		return errors.Trace(err)
 	}
-
-	info := do.infoHandle.Get()
-	if info != nil && schemaMetaVersion <= info.SchemaMetaVersion() {
-		// info may be changed by other txn, so here its version may be bigger than schema version,
-		// so we don't need to reload.
-		log.Debugf("[ddl] schema version is still %d, no need reload", schemaMetaVersion)
+	m := meta.NewSnapshotMeta(snapshot)
```

Comment:
I tried to extract create meta outside of this function, but it adds many lines of code.
create meta inside, so the caller doesn't need to care about meta.